### DISCUSSION TITLE
Maintain git version updated automatically ?

### DISCUSSION THREAD

**Discussion by zbugrkx** [acassen/keepalived#2516] [Q&A]
Hi!

To stay up to date with fixes etc, I moved, as per recommendations from acassen to pulling/building the git version. 

Works much better than the snap variant!

My question now is, since this is a homelab system anyway, i can take the risk, what would be the best/easiest way to get my pi to check for updates on git and pull them? As I would prefer to stay updated versus checking manually from time to time.

Cronjob weekly with this: ?

git -C /directory/keepalived pull 
sudo make -j6
sudo cp /directory/keepalived/keepalived /usr/sbin
sudo cp /directory/keepalived/doc/*-MIB.txt /usr/share/snmp.mibs

sudo systemctl restart keepalived

Thanks for any advice !

**Comment by pqarmitage** [comment]
The make command needs to be executed in the top level directory of the git tree (where the configure.ac file is). I think the paths you have given in the commands are inconsistent, so I suggest the following:
```
cd directory             # this is the directory with configure.ac in it
git pull
make -j6                   # this does not need to be executed with sudo *
sudo cp keepalived/keepalived /usr/sbin
sudo cp doc/*-MIB.txt /usr/share/snmp/mibs     # note: .../snmp/mibs, not .../snmp.mibs
```
* if you have already built keepalived using root privileges, you will have some build-time generated files owned by root. To clear these (and I am assuming that you didn't execute the git clone as root), and it just needs doing once, after `cd directory` execute `sudo make distclean` and then you will need to run the configure command again. If you did execute the git clone as root, then `cd directory/..; rm -rf keepalived`, then as a non-privileged user execute the git clone command again.

**Comment by zbugrkx** [comment]
Thanks!

I did some testing a bit after posting this but did not have time to come back and edit. I had added a cd / into my directory as it made the script a whole lot easier/better to work.

As for the make command, did not want to bother you on the other thread but when I built it yesterday, it would not work. When I ran the make command with sudo, everything went further and I was able to run keepalived. Unsure if there is maybe some issues on my side but, unless this can cause a big issue, i guess I can keep it that way?

**Comment by pqarmitage** [comment]
If make won't run as an ordinary user, then something is wrong. It might be that you have a file in the build tree that is owned by root, or by a user other than the one who is trying to build keepalived, and therefore don't have permission to delete or update the file. It shouldn't cause a problem build keepalived as root, but on the other hand something is wrong if you have to and it would be better to sort that out.


Perhaps it might be easiest to start again. So logged is as your normal, non-root, user:

> cd ~
git clone https://github.com/acassen/keepalived
cd keepalived
./configure --enable-snmp --enable-sha1 --enable-snmp-rfcv2 --enable-snmp-rfcv3 --enable-dbus --enable-json --enable-bfd --enable-regex --with-init=systemd
./autogen.sh
make -j6

If that works you can remove you original /directory. If it doesn't work as a non-root user please post the error message(s) you are getting.

**Reply by zbugrkx** [reply]
Thank you very much!

I think you are right and it might have been that the directory was owned by root, my mistake.

I followed your steps above to restart, beside having to run autogen before configure, else it would fail, everything worked without need to sudo except the last part of copying the files and enable/restart wity systemctl as expected !

I'll update my script file in cron and I should be all set!

Many thanks again :-)
